src/hooks/core/useCoreChatSystemHook.ts

```diff
@@ -171,21 +172,32 @@ const useCoreChatSystemHook = () => {
 
     if (isSystemMessage && isContainFollowingMessage) {
       const textOwnUser = 'You started following this user.\n Send them a message now.';
-      queue.addPriorityJob({
-        operationLabel: DatabaseOperationLabel.CoreChatSystem_SaveFollowingMessage,
-        id: `${websocketData?.channel?.id}-${websocketData?.message?.id}`,
-        priority: QueueJobPriority.HIGH,
-        task: async () => {
-          await ChannelList.updateChannelDescription(
-            localDb,
-            websocketData?.channel_id,
-            websocketData?.message?.textOwnMessage ?? textOwnUser,
-            websocketData
-          );
-          const chat = ChatSchema.fromWebsocketObject(websocketData);
-          await chat.save(localDb);
-        }
-      });
+      if (Platform.OS === 'android') {
+        await ChannelList.updateChannelDescription(
+          localDb,
+          websocketData?.channel_id,
+          websocketData?.message?.textOwnMessage ?? textOwnUser,
+          websocketData
+        );
+        const chat = ChatSchema.fromWebsocketObject(websocketData);
+        await chat.save(localDb);
+      } else {
+        queue.addPriorityJob({
+          operationLabel: DatabaseOperationLabel.CoreChatSystem_SaveFollowingMessage,
+          id: `${websocketData?.channel?.id}-${websocketData?.message?.id}`,
+          priority: QueueJobPriority.HIGH,
+          task: async () => {
+            await ChannelList.updateChannelDescription(
+              localDb,
+              websocketData?.channel_id,
+              websocketData?.message?.textOwnMessage ?? textOwnUser,
+              websocketData
+            );
+            const chat = ChatSchema.fromWebsocketObject(websocketData);
+            await chat.save(localDb);
+          }
+        });
+      }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The changes here introduce a conditional check for Android and other platforms. For Android, tasks are executed directly, while for other platforms, tasks are added to a priority job queue. This approach could help improve performance on Android by avoiding unnecessary queuing. However, it's important to ensure that these direct operations do not block the UI thread and cause jank. If they do, it might be better to use a job queue or worker thread even on Android.
```diff
+ if (Platform.OS === 'android') {
+   await ChannelList.updateChannelDescription(
+     localDb,
+     websocketData?.channel_id,
+     websocketData?.message?.textOwnMessage ?? textOwnUser,
+     websocketData
+   );
+   const chat = ChatSchema.fromWebsocketObject(websocketData);
+   await chat.save(localDb);
+ } else {
+   queue.addPriorityJob({
+     operationLabel: DatabaseOperationLabel.CoreChatSystem_SaveFollowingMessage,
+     id: `${websocketData?.channel?.id}-${websocketData?.message?.id}`,
+     priority: QueueJobPriority.HIGH,
+     task: async () => {
+       await ChannelList.updateChannelDescription(
+         localDb,
+         websocketData?.channel_id,
+         websocketData?.message?.textOwnMessage ?? textOwnUser,
+         websocketData
+       );
+       const chat = ChatSchema.fromWebsocketObject(websocketData);
+       await chat.save(localDb);
+     }
+   });
+ }
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->